compiler/rustc_trait_selection/src/opaque_types.rs

```diff
@@ -691,26 +693,26 @@ impl<'tcx, OP> TypeVisitor<'tcx> for ConstrainOpaqueTypeRegionVisitor<OP>
 where
     OP: FnMut(ty::Region<'tcx>),
 {
-    fn visit_binder<T: TypeFoldable<'tcx>>(&mut self, t: &ty::Binder<T>) -> bool {
+    fn visit_binder<T: TypeFoldable<'tcx>>(&mut self, t: &ty::Binder<T>) -> ControlFlow<(), ()> {
         t.as_ref().skip_binder().visit_with(self);
```

Comment:
`ControlFlow` should probably be `#[must_use]` so we'd be forced to use `let _ = ` here, but that's for another PR

Reply:
Yeah, making it `#[must_use]` seems like a good idea to me

Reply:
Opened #78202

Reply:
Another thing you could consider to avoid the conversation on the type would be to just mark the methods on the trait as `#[must_use]`.  It [looks](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=da1fcb2728ab2129196d98af35b367d9) like that works even for calls on concrete things that don't copy the attribute to their implementation.  Probably still not this PR, though.